plugins/modules/device_credential_intent.py

```diff
@@ -2407,6 +2411,145 @@ def get_diff_deleted(self, config):
 
         return self
 
+    def verify_diff_merged(self, config):
+        """
+        Validating the DNAC configuration with the playbook details
+        when state is merged (Create/Update).
+
+        Parameters:
+            config (dict) - Playbook details containing Global Pool,
+            Reserved Pool, and Network Management configuration.
+
+        Returns:
+            self
+        """
+
+        self.log(str("Entered the verify function."))
+        self.get_have(config)
+        self.get_want(config)
+        self.log("DNAC retrieved details: " + str(self.have))
+        self.log("Playbook details: " + str(self.want))
+        if config.get("global_credential_details") is not None:
+            if self.want.get("want_create"):
+                self.msg = "Global Device Credentials config is not applied to the DNAC"
+                self.status = "failed"
+                return self
+            if self.want.get("want_update"):
```

Comment:
Can we optimize this part? You can add all credentials in a list and do it.. 

For values, you can extend based on the credential type..
```
                    values = ["description", "id"]
                    if credential_type in ["httpsRead", "httpsWrite"]:
                        values.extend(["username", "port"])
                    elif credential_type in ["cliCredentia"]:
                        values.extend(["username"])
                    elif ..

```

                   
Sample Code like below:
```
        if self.want.get("want_update"):
            credential_types = ["cliCredential", "snmpV2cRead", "snmpV2cWrite", "httpsRead", "httpsWrite", "snmpV3"]
            for credential_type in credential_types:
                if self.want.get(credential_type):
                    want_credential = self.want.get(credential_type)
                    have_credential = self.have.get(credential_type)

                    if not have_credential:
                        self.msg = f"{credential_type} Credentials config is not applied to the DNAC"
                        self.status = "failed"
                        return self

                    values = ["description", "id"]
                    if credential_type in ["httpsRead", "httpsWrite"]:
                        values.extend(["username", "port"])
                   

                    for value in values:
                        equality = have_credential.get(value) == want_credential.get(value)
                        if not equality:
                            self.msg = f"{credential_type} Credentials config is not applied to the DNAC"
                            self.status = "failed"
                            return self
```

Reply:
Or simply we can map it into dictionary and use it.


```
                    value_mapping = {
                        "cliCredential": ["username", "description", "id"],
                        "snmpV2cRead": ["description", "id"],
                        "snmpV2cWrite": ["description", "id"],
                        "httpsRead": ["description", "username", "port", "id"],
                        "httpsWrite": ["description", "username", "port", "id"],
                        "snmpV3": ["username", "description", "snmpMode", "id"]
                    }

                    values = value_mapping.get(credential_type, [])
                    for value in values:
                        equality = have_credential.get(value) == want_credential.get(value)
                        if not equality:
                            self.msg = f"{credential_type} Credentials config is not applied to the DNAC"
                            self.status = "failed"
                            return self
```